sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -3185,11 +3266,13 @@ pi_result piQueueRelease(pi_queue Queue) {
         if (it->second.InUse) {
           Queue->resetCommandList(it, true);
         }
-        // TODO: remove "if" when the problem is fixed in the level zero
-        // runtime. Destroy only if a queue is healthy. Destroying a fence may
-        // cause a hang otherwise.
-        if (Queue->Healthy)
-          ZE_CALL(zeFenceDestroy, (it->second.ZeFence));
+        // Immediate commandlists do not have a fence.
+        if (!UseImmediateCommandLists)
```

Comment:
Let's just check if fence != nullptr (and document that nullptr identifies immediate command-list)

Reply:
OK, done.